File: File Sorter/random_txt_files/sorterWithTime.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <dirent.h>
#include <time.h>   // for timing

int main() {
    DIR *folder;
    struct dirent *entry;
    char zip_name[100];
    char command[1024] = "zip ";
    int count = 0;

    printf("Enter a name for your zip file (without .zip): ");
    scanf("%99s", zip_name);

    if (strstr(zip_name, ".zip") != 0) {
        zip_name[strlen(zip_name) - 4] = '\0';
        printf("You included '.zip', it will be removed.\n");
    }

    strcat(command, zip_name);
    strcat(command, ".zip");

    folder = opendir(".");
    if (folder == 0) {
        printf("Could not open current directory.\n");
        return 1;
    }

    while ((entry = readdir(folder)) != 0) {
        char *ext = strrchr(entry->d_name, '.');
        if (ext && strcmp(ext, ".txt") == 0) {
            strcat(command, " ");
            strcat(command, entry->d_name);
            count++;
        }
    }
    closedir(folder);

    if (count > 0) {
        clock_t start_time = clock();    // Start timing before zipping
        system(command);                 // Run the zip command
        clock_t end_time = clock();      // End timing after zipping
        double elapsed_time = (double)(end_time - start_time) / CLOCKS_PER_SEC;

        // ✅ Final output
        printf("There are number of %d .txt files and compressed into a .zip file\n", count);
        printf("Created zip file path: ./%s.zip\n", zip_name);
        printf("Zipping execution time: %.6f seconds\n", elapsed_time);
    } else {
        printf("No .txt files found in this folder.\n");
    }

    return 0;
}

File: File Sorter/random_txt_files/sorterWithTime.py
# Import necessary modules
import os           # For interacting with the operating system (file paths, directory listings, etc.)
import zipfile      # For creating zip files
import time         # For measuring execution time

# Get current working directory
current_dir = os.getcwd()

# Ask user for zip file name
zip_name = input("Enter a name for your zip file (without .zip): ").strip()

# Handle empty input
if zip_name == "":
    zip_name = "mytxt"
    print("No name entered. Using default name: mytxt.zip")
# Remove ".zip" if the user included it
elif zip_name.lower().endswith(".zip"):
    zip_name = zip_name[:-4].strip()
    print("Removed '.zip' from your input to avoid duplication.")

# Create the zip file path
zip_path = os.path.join(current_dir, zip_name + ".zip")

# Find all .txt files in current directory
txt_files = [
    f for f in os.listdir(current_dir)
    if f.lower().endswith(".txt") and os.path.isfile(os.path.join(current_dir, f))
]


count = len(txt_files)

# Compress if there are .txt files
if count > 0:
    start_time = time.perf_counter()

    with zipfile.ZipFile(zip_path, "w", zipfile.ZIP_DEFLATED) as zipf:
        for f in txt_files:
            zipf.write(f, arcname=os.path.basename(f))

    end_time = time.perf_counter()
    elapsed = end_time - start_time

    # ✅ Final output (matches your C version)
    print(f"There are number of {count} .txt files and compressed into a .zip file")
    print(f"Created zip file path: {zip_path}")
    print(f"Zipping execution time: {elapsed:.6f} seconds")

else:
    print("No .txt files found in this directory.")
